src/RA_Dlg_Login.cpp

```diff
@@ -34,76 +38,46 @@ INT_PTR CALLBACK RA_Dlg_Login::RA_Dlg_LoginProc(HWND hDlg, UINT uMsg, WPARAM wPa
             {
                 case IDOK:
                 {
-                    BOOL bValid = TRUE;
                     TCHAR sUserEntry[64];
-                    bValid &= (GetDlgItemText(hDlg, IDC_RA_USERNAME, sUserEntry, 64) > 0);
-                    TCHAR sPassEntry[64];
-                    bValid &= (GetDlgItemText(hDlg, IDC_RA_PASSWORD, sPassEntry, 64) > 0);
+                    if (GetDlgItemText(hDlg, IDC_RA_USERNAME, sUserEntry, 64) == 0)
+                    {
+                        ra::ui::viewmodels::MessageBoxViewModel::ShowErrorMessage(L"Username is required.");
+                        return TRUE;
+                    }
 
-                    if (!bValid || lstrlen(sUserEntry) < 0 || lstrlen(sPassEntry) < 0)
+                    TCHAR sPassEntry[64];
+                    if (GetDlgItemText(hDlg, IDC_RA_PASSWORD, sPassEntry, 64) == 0)
                     {
-                        MessageBox(nullptr, TEXT("Username/password not valid! Please check and reenter"), TEXT("Error!"), MB_OK);
+                        ra::ui::viewmodels::MessageBoxViewModel::ShowErrorMessage(L"Password is required.");
                         return TRUE;
                     }
 
-                    PostArgs args;
-                    args['u'] = ra::Narrow(sUserEntry);
-                    args['p'] = ra::Narrow(sPassEntry);		//	Plaintext password(!)
+                    ra::api::Login::Request request;
+                    request.Username = ra::Narrow(sUserEntry);
+                    request.Password = ra::Narrow(sPassEntry);
 
-                    rapidjson::Document doc;
-                    if (RAWeb::DoBlockingRequest(RequestLogin, args, doc))
+                    const ra::api::Login::Response response = request.Call();
+
+                    if (response.Succeeded())
                     {
-                        std::string sResponse;
-                        std::string sResponseTitle;
-
-                        if (doc["Success"].GetBool())
-                        {
-                            const std::string& sUser = doc["User"].GetString();
-                            const std::string& sToken = doc["Token"].GetString();
-                            const unsigned int nPoints = doc["Score"].GetUint();
-                            const unsigned int nUnreadMessages = doc["Messages"].GetUint();
```

Comment:
extracting the JSON values is done in the API call, which ensures the fields are handled consistently no matter where the API is called from. In this case, Score and Messages are optional and this code doesn't check for that.